app/src/main/java/com/cornellappdev/scoop/ui/viewmodel/HomeScreenViewModel.kt

```diff
@@ -0,0 +1,20 @@
+package com.cornellappdev.scoop.ui.viewmodel
+
+import androidx.lifecycle.ViewModel
+import com.cornellappdev.scoop.data.repositories.LoginRepository
+import com.cornellappdev.scoop.data.repositories.RideRepository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+
+@HiltViewModel
+class HomeScreenViewModel @Inject constructor(
+    private val rideRepository: RideRepository,
+) : ViewModel() {
+
+    val rideFlow = rideRepository.rideFlow
+
+    suspend fun getRides() {
+        rideRepository.getAllRides()
+    }
```

Comment:
I'm usually a proponent of turning these `suspend fun`s into regular `fun`s that just launch a `viewModelScope` coroutine to do whatever suspending stuff they do. Makes it way easier to call in the UI components. (Also, add a spec!)